project/CodeStone1125/README.md

```diff
@@ -0,0 +1,182 @@
+# 2D Collision Detector
+
+
+## Basic Information
+
+Simulate how different type of shape collide in a 2D map.
+
+* github repository: [https://github.com/CodeStone1125/2DCollisionDetector](https://github.com/CodeStone1125/2DCollisionDetector)
+
+## Problem to Solve
+
+The objective of this system is to calculate scenarios for the most common shapes in
+a 2D map, such as rectangles, circles, and polygons. Before we dive deeper into the 
+topic, let's first discuss the "Axis-Aligned Bounding Box" (AABB) algorithm. While 
+it's not a single algorithm but a framework for collision detection, it forms the basis
+for many collision detection systems in 2D graphics and game development due to its 
+simplicity and efficiency. Moreover, in the 2D world, AABB can effectively handle 
+approximate collision detection for most cases.
+
+However, for more complex polygons, we need to introduce another algorithm called the
+Separating Axis Theorem (SAT). SAT is more precise but also more resource-intensive.
+For most cases in the 2D world, we canhandle collision detection through the following
+steps:
+
+1. Preliminary screening of collision objects using AABB.
+2. Further checking with SAT.
+
+
+### AABB's Pseudo code
+* Bounding Boxes: Each object in a 2D scene is enclosed by a rectangle (a bounding box)
+   that is aligned with the axes of the coordinate system.These rectangles are often
+   referred to as "AABBs" because they are aligned with the axes.
+   Here is the pseudo code in `python`:
+  
+  ```python
+  def calculate_aabb(points):
```

Comment:
Collision detection of two boxes is extremely simple, needing little calculation.  It is so simple that coding it in Python would probably be fast enough.

It will be really hard for you to make a decent presentation using only that calculation.